platform/core/PolyPodCoreAndroid/core/src/main/kotlin/coop/polypoly/core/JniApi.kt

```diff
@@ -1,10 +1,90 @@
 package coop.polypoly.core
 
+import org.msgpack.core.MessagePack
+import org.msgpack.core.MessageUnpacker
+import org.msgpack.value.ValueFactory
+import java.io.ByteArrayOutputStream
+
+enum class NativeRequest {
+    Example
+}
+
+sealed interface NativeResponse<T> {
+    fun response(): T
+    fun messagePackedAsOk(): ByteArray
+}
+
+object Example : NativeResponse<String> {
+    override fun response(): String {
+        return "Test"
+    }
+
+    override fun messagePackedAsOk(): ByteArray {
+        val map = ValueFactory.newMap(
+            mutableMapOf(
+                ValueFactory.newString("Ok") to ValueFactory.newMap(
+                    mutableMapOf(
+                        ValueFactory.newString(
+                            "Example"
+                        ) to ValueFactory.newString(
+                            response()
+                        )
+                    )
+                )
+            )
+        )
+        val output = ByteArrayOutputStream()
+        val packer = MessagePack.newDefaultPacker(output)
+        packer.packValue(map)
+        packer.close()
+        return output.toByteArray()
+    }
+}
+
 class JniApi {
     init {
         System.loadLibrary("polypod_core")
     }
 
-    external fun bootstrapCore(languageCode: String): ByteArray
+    private fun mapToNativeRequest(request: String): NativeRequest {
+        return NativeRequest.valueOf(request)
+    }
+
+    private fun handle(nativeRequest: NativeRequest): NativeResponse<Any> {
+        when (nativeRequest) {
+            NativeRequest.Example -> return Example as NativeResponse<Any> // ktlint-disable max-line-length
+        }
+    }
+
+    private fun packExp(exception: Exception): ByteArray {
+        val map = ValueFactory.newMap(
+            mutableMapOf(
+                ValueFactory.newString("Err") to ValueFactory.newString(
+                    exception.toString()
+                )
+            )
+        )
+        val output = ByteArrayOutputStream()
+        val packer = MessagePack.newDefaultPacker(output)
+        packer.packValue(map)
+        packer.close()
+        return output.toByteArray()
+    }
+
+    fun performRequest(input: ByteArray): ByteArray {
+        return try {
+            val unpacker: MessageUnpacker = MessagePack.newDefaultUnpacker(
+                input
+            )
+            val request = unpacker.unpackValue().asStringValue().toString()
```

Comment:
This will not always be a string, maybe move the decision to mapToNativeRequest function, so we don't have to ever touch this function again.